external-crates/move/move-compiler/src/diagnostics/mod.rs

```diff
@@ -416,24 +419,29 @@ impl WarningFilters {
             WarningFilters::All => (),
             WarningFilters::Empty => {
                 *self = WarningFilters::Specified {
-                    category: BTreeSet::new(),
-                    codes: BTreeSet::new(),
+                    category: BTreeMap::new(),
+                    codes: BTreeMap::new(),
                 };
                 return self.add(filter);
             }
             WarningFilters::Specified { category, codes } => match filter {
                 WarningFilter::All => *self = WarningFilters::All,
-                WarningFilter::Category(cat) => {
+                WarningFilter::Category(cat, n) => {
                     let cat = cat as u8;
-                    category.insert(cat);
+                    category.insert(cat, n);
                     // remove any codes now covered by this category
-                    codes.retain(|(codes_cat, _)| codes_cat != &cat);
+                    codes.retain(
+                        |DiagnosticsID {
+                             category: codes_cat,
+                             code: _,
+                         },
+                         _| codes_cat != &cat,
```

Comment:
```suggestion
                        |diag_id, _| diag_id.cateogry != &cat,
```
Love the pattern matching :) But this might be a bit more readable